admin_guide/scheduling/priority_preemption.adoc

```diff
@@ -0,0 +1,138 @@
+[[admin-guide-priority-preemption]]
+= Pod Priority and Preemption
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+Pods can have priority. Priority indicates the importance of a Pod relative to other Pods. If a Pod cannot be scheduled, the scheduler tries to preempt (evict) lower priority Pods to make scheduling of the pending Pod possible.
+
+Priority also affects scheduling order of Pods and out-of-resource eviction ordering on the Node.Pod priority and preemption .
+
+== How to use priority and preemption
+
+To use priority and preemption, follow these steps:
+
+. Add one or more PriorityClasses.
+
+. Create Pods with priorityClassName set to one of the added PriorityClasses. Of course you do not need to create the Pods directly; normally you would add priorityClassName to the Pod template of a collection object like a Deployment.
+
+
+If you try the feature and then decide to disable it, you must remove the PodPriority command-line flag or set it to false, and then restart the API server and scheduler. After the feature is disabled, the existing Pods keep their priority fields, but preemption is disabled, and priority fields are ignored. If the feature is disabled, you cannot set priorityClassName in new Pods
+
+== How to use priority and preemption
+
+Preemption is controlled by a scheduler flag disablePreemption, which is set to false by default.
+
+It is important to note that critical pods (except DaemonSet pods, which are still scheduled by the DaemonSet controller) rely on scheduler preemption to be scheduled when a cluster is under resource pressure. For this reason, we do not recommend disabling preemption.
+
+== PriorityClass
+
+A PriorityClass is a non-namespaced object that defines a mapping from a priority class name to the integer value of the priority. The name is specified in the name field of the PriorityClass object’s metadata. The value is specified in the required value field. The higher the value, the higher the priority.
+
+A PriorityClass object can have any 32-bit integer value smaller than or equal to 1 billion. Larger numbers are reserved for critical system Pods that should not normally be preempted or evicted. We have two reserved priorityClasses for critical system Pods to have guaranteed scheduling.
+
```

Comment:
@ravisantoshgudimetla Does the user need to add one of these priority classes to critical pods? Is it done automatically? For new critical pods and upgrade scenarios.